pyccel/codegen/pipeline.py

```diff
@@ -196,12 +212,65 @@ def handle_error(stage):
-    #            pass
+        # if recursive:
+        #     for dep in parser.sons:
+        #         # Call same function on 'dep'
+        #         pass
         #------------------------------------------------------
 
+        # Iterate over the internal_libs list and determine if the printer
+        # requires an internal lib to be included.
+        for lib in internal_libs:
+            if lib in codegen.get_printer_imports():
+                # get the include folder path and library files
+                if lib not in internal_libs_name:
+                    # get the library folder name
+                    lib_name = internal_libs[lib]
+                    # get path to pyccel/stdlib/lib_name
+                    import pyccel.stdlib as stdlib_folder
+                    stdlib_path = os.path.dirname(stdlib_folder.__file__)
+                    lib_path = os.path.join(stdlib_path, lib_name)
+                    # remove library folder to avoid missing files and copy
+                    # new one from pyccel stdlib
+                    lib_dest_path = os.path.join(pyccel_dirpath, lib_name)
+                    if os.path.exists(lib_dest_path):
+                        shutil.rmtree(lib_dest_path)
+                    shutil.copytree(lib_path, lib_dest_path)
+
+                    # stop after copying lib to __pyccel__ directory for
+                    # convert only
+                    if convert_only:
+                        continue
+
+                    # get library source files
+                    source_files = []
+                    for e in os.listdir(lib_dest_path):
+                        if e.endswith(lang_ext_dict[language]):
+                            source_files.append(os.path.join(lib_dest_path, e))
+
+                    # compile library source files
+                    flags = construct_flags(f90exec,
+                                            fflags=fflags,
+                                            debug=debug,
+                                            includes=[lib_dest_path])
+                    try:
+                        for f in source_files:
+                            compile_files(f, f90exec, flags,
+                                            binary=None,
+                                            verbose=verbose,
+                                            is_module=True,
+                                            output=lib_dest_path,
+                                            language=language)
+                    except Exception:
+                        handle_error('C {} library compilation'.format(lib))
+                        raise
+                    # Add internal lib to internal_libs_name dict
+                    internal_libs_name[lib] = 1
+                    # add source file without extension to internal_libs_files
+                    internal_libs_files.extend([os.path.splitext(f)[0] for f in source_files])
```

Comment:
You don't need to convert the generator expression to a list before calling extend. You can just do:
```
internal_libs_files.extend(os.path.splitext(f)[0] for f in source_files)
```

See: https://www.geeksforgeeks.org/python-list-comprehensions-vs-generator-expressions/

Reply:
ok, I will change it